### PR TITLE
Fix crash caused by malformed prompt or title

### PR DESCRIPTION
**Pull request by mitnk**
Following @adqm 's fixing prompt in #1150 , this PR try to fix bad `$TITLE` and `$PROMPT` from callables.

The following prompt in xonshrc was causing xonsh crash:

```
def render_prompt():
    return '{HELLO'
$PROMPT = render_prompt
```

@adqm @scopatz please help to review it, thanks.

### CODE REVIEW

File: xonsh/environ.py [hunk 1]
```diff
@@ -1135,7 +1135,26 @@ def _get_fmtter(formatter_dict=None):
...
 
+def _failover_template_format(template):
+    if callable(template):
+        try:
+            # Exceptions raises from function of producing $PROMPT
+            # in user's xonshrc should not crash xonsh
+            return template()
+        except:
+            print_exception()
+            return '$ '
```
Comment (mitnk) [inline xonsh/environ.py]: any suggestions here?

File: xonsh/environ.py [hunk 2]
```diff
@@ -1135,7 +1135,26 @@ def _get_fmtter(formatter_dict=None):
     return fmtter
 
 
+def _failover_template_format(template):
+    if callable(template):
+        try:
+            # Exceptions raises from function of producing $PROMPT
+            # in user's xonshrc should not crash xonsh
+            return template()
+        except:
```
Comment (scopatz) [inline xonsh/environ.py]: This should be `except Exception:` so it doesn't catch keyboard interrupts

File: xonsh/environ.py (hunk 1 above)
Comment (scopatz) [inline xonsh/environ.py]: This seems fine to me!

**Comment by scopatz** [comment]
This looks good to me! Thanks for putting this in @mitnk! Just a couple of small issue.  Also thanks a ton for putting in the tests!

**Comment by mitnk** [comment]
updated except, thanks

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**